tests/qnn/test_keras.py

```diff
@@ -542,3 +542,35 @@ def test_model_save_weights(self, model, n_qubits, tmpdir):
         assert np.allclose(prediction, prediction_loaded)
         for i, w in enumerate(weights):
             assert np.allclose(w, weights_loaded[i])
+
+    @pytest.mark.parametrize("n_qubits, output_dim", indices_up_to(1))
+    def test_model_serialization(self, get_circuit, output_dim):
+        """Test if serializing and deserializing `KerasLayer` as part of a 
+        Keras model reconstructs the layer correctly, i.e. that the right
+        arguments are passed to __init__ when instantiating the layer during
+        deserialization."""
+        c, w = get_circuit
+        layer = KerasLayer(c, w, output_dim)
+
+        inputs = tf.keras.layers.Input((1,))
+        outputs = layer(inputs)
+
+        keras_model = tf.keras.Model(inputs=inputs, outputs=outputs)
+        model_clone = tf.keras.models.clone_model(keras_model)
+
+        layer_clone = model_clone.layers[-1]
+
+        # Assert that the KerasLayer in the deserialized model has been 
+        # instantiated with the correct __init__ args.
+
+        # Test for identity (versus equality) here to make sure that the cloned
+        # model is evaluated on the same device
+        assert layer.qnode is layer_clone.qnode
```

Comment:
I'm not sure if this should be expected behaviour or not :thinking:

Reply:
I see your point... What approach would you suggest here? Test for equality for now and perhaps modify the test case once qnode serialization is implemented?

Reply:
Well, this seems to indicate that we'll be referencing the same QNode with both layers. Is that something we'd want? Are there cases where evaluating the qnode of one layer would mutate that QNode and hence affect how the QNode of the other layer behaves? If so, we'd like to copy the QNode and make a deep copy of the underlying objects while creating the clone of the layer.

On another note, to check equality, we could test two things:

* Check that the two QNodes have the same operations in their underlying `QuantumTape`, by iterating through `layer.qnode.qtape.operations` and ``layer_clone.qnode.qtape.operations` and asserting the ops,
* Check that the underlying device tied with the QNodes are the same.